### PR TITLE
15985-allowable-states-for-limited-restoration-corps

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity#15985

In the case of a corporation undergoing Limited Restoration, the user is able to file Change of Address, Change of Directors, and Annual Report, but is unable to file Alteration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/authz.py [hunk 1]
```diff
@@ -343,7 +343,8 @@ def get_allowable_filings_dict():
                 'alteration': {
                     'legalTypes': ['BC', 'BEN', 'ULC', 'CC'],
                     'blockerChecks': {
-                        'business': [BusinessBlocker.DEFAULT]
+                        'business': [BusinessBlocker.DEFAULT],
+                        'validStateFilings': ['limitedRestoration', 'limitedRestorationExtension']
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/authz.py:347]: This wont work. It says a user can only file an alteration when its in limitedRestoration or limitedRestorationExtension

File: legal-api/tests/unit/services/test_authorization.py [hunk 2]
```diff
@@ -1851,6 +1865,7 @@ def mock_auth(one, two):  # pylint: disable=unused-argument; mocks of library me
          ['dissolution', None], [None, None], [])
     ]
 )
+#James - Add test cases here
```
Comment (vysakh-menon-aot) [inline legal-api/tests/unit/services/test_authorization.py:1868]: you can remove this

File: legal-api/tests/unit/services/test_authorization.py [hunk 3]
```diff
@@ -1773,6 +1779,14 @@ def mock_auth(one, two):  # pylint: disable=unused-argument; mocks of library me
                           FilingKey.REGISTRARS_ORDER])),
 
         # active business - general user
+        ('general_user_cp', Business.State.ACTIVE, ['CP'], 'general', [BASIC_USER],
```
Comment (vysakh-menon-aot) [inline legal-api/tests/unit/services/test_authorization.py:1782]: Can you change this to BEN or BC. CP does not have alteration filing

**Comment by vysakh-menon-aot** [comment]
Can you fix the lint error

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by PaulGarewal — APPROVED** [review]

**Pull request merged**